qiskit/utils/quantum_instance.py

```diff
@@ -480,8 +480,13 @@ def execute(self, circuits, had_transpiled: bool = False):
             build_measurement_error_mitigation_qobj,
         )
 
-        # maybe compile
-        if not had_transpiled:
+        if had_transpiled:
+            if isinstance(circuits, list):
+                circuits = circuits.copy()
```

Comment:
Just a minor thing - I wonder whether it might be an idea to state why the copy is done here in a comment for future reference so that the reason is know and no-one might be tempted to remove the copy thinking they are improving performance. I guess the copy is only needed when we are doing measurement mitigation that will add to the list of circuits being executed. Since this is a shallow copy of the list I guess it should be fine having this done all the time and not dependent on when we know we would add extra ccts.

Reply:
I will add a comment. Yes, currently the error is just when using measurement mitigation but who knows someone could add code to change the input array somewhere else in the future. The method should not be changing the input without mentioning it in the documentation in any situation.

Reply:
If this does only a shallow copy, is it what we want? If we go on modifying the shallow copied `circuits` list, then the original objects will also be changed, right?
```python
from qiskit.circuit.library import EfficientSU2

original = EfficientSU2(2, reps=1)
circuits = [original]

circuits = circuits.copy()
circuits[0].x(0)

print(original.draw())  # has a new X gate
```

Reply:
No, the QI code never changes the original circuit. In fact QI expects it to be an array and was changing the array in place in the mitigation part. There is only the need to make shallow copy to avoid it. A deep copy is overkill and would slow the code.
Making just a shallow copy was intentional.

Reply:
Ah, ok now I understand. Thanks! 🙂